ui/shared/DropdownMenuItems.jsx

```diff
@@ -4,39 +4,60 @@ import { DropdownMenu, DropdownItem } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faCheck } from '@fortawesome/free-solid-svg-icons';
 
+const createDropdownItem = (item, namespace, updateData, selectedItem) => {
+  return (
+    <DropdownItem
+      tag="a"
+      key={`${namespace}${item}`}
+      onClick={() => updateData(item)}
+    >
+      <FontAwesomeIcon
+        icon={faCheck}
+        className={`mr-1 ${selectedItem === item ? '' : 'hide'}`}
+        title={selectedItem === item ? 'Checked' : ''}
+      />
+      {item}
+    </DropdownItem>
+  );
+};
+
 const DropdownMenuItems = ({
   selectedItem,
   updateData,
   options,
+  pinned,
   namespace = '',
 }) => (
   <DropdownMenu className="overflow-auto dropdown-menu-height">
-    {options.map(item => (
-      <DropdownItem
-        tag="a"
-        key={`${namespace}${item}`}
-        onClick={() => updateData(item)}
-      >
-        <FontAwesomeIcon
-          icon={faCheck}
-          className={`mr-1 ${selectedItem === item ? '' : 'hide'}`}
-          title={selectedItem === item ? 'Checked' : ''}
-        />
-        {item}
-      </DropdownItem>
-    ))}
+    {/* Items pinned to top of dropdown */}
+    {pinned.length > 0 &&
+      options
+        .filter(item => pinned.includes(item))
+        .sort((a, b) => a > b)
+        .map(item =>
+          createDropdownItem(item, namespace, updateData, selectedItem),
+        )}
+    {pinned.length > 0 && <DropdownItem divider />}
```

Comment:
I see an edge case here: what if we have 5 pinned items, but none are found in `options`?
Basically, we'll have no pinned items, followed by a horizontal dash, followed by `options`.

We need to address this small issue as well.

Reply:
good catch, thanks!

Reply:
a thought: filter the `pinned` array to only pass values that also appear in `options`. Tested this by changing `options` to contain only `['bogus']`, and confirmed that only 'bogus' appear in the drop down, with no divider

Reply:
This works, but we could make it a bit more future proof.
As it currently is, this PR treats pinned items in a generic way. `DropdownMenuItems` component helps a lot here.
If we would actually move that filtering from `TestDataModal` *(client)* code to `DropdownMenuItems`, we could ensure that all dropdown menu are able to handle pinned items, not just project-based dropdowns.

Reply:
Isn't the` DropdownMenuItems` component already filtering based on the `pinned` props? This code looks like it's good enough for now. People can always modify a component in the future if the need arises.

Reply:
alrighty, thanks @sarah-clements